### PR TITLE
subsidies: draw down a balance instead of minting

### PR DESCRIPTION
**Pull request by bmwill**
In an effort to eliminate the ability to mint arbitrary amounts of Sui, we'll have the StakeSubsidy structure custody the total balance of tokens set asside for subsidies and drawn down that balance over time instead of minting tokens.

## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] user-visible impact
- [x] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [x] breaking change for on-chain data layout
- [x] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-framework/sources/sui.move [hunk 1]
```diff
@@ -6,18 +6,28 @@
...
+    /// The total supply of Sui denominated in Mist (10 Billion * 10^9)
+    const TOTAL_SUPPLY_MIST: u64 = 10000000000000000000;
+
     /// Name of the coin
     struct SUI has drop {}
 
     /// Register the `SUI` Coin to acquire its `Supply`.
     /// This should be called only once during genesis creation.
-    public(friend) fun new(ctx: &mut TxContext): Supply<SUI> {
+    public(friend) fun new(ctx: &mut TxContext): Balance<SUI> {
```
Comment (sblackshear) [inline crates/sui-framework/sources/sui.move:30]: A
```
assert!(tx_context::epoch_timestamp(ctx) == 0, EAlreadyMinted)
```

(assuming epoch time starts at 0) might be a good defense-in-depth mechanism here.

**Review by sblackshear — APPROVED** [review]
Very glad to see this!!

File: crates/sui-framework/sources/governance/genesis.move [hunk 2]
```diff
@@ -25,6 +27,9 @@ module sui::genesis {
     /// Stake subisidy to be given out in the very first epoch. Placeholder value.
     const INIT_STAKE_SUBSIDY_AMOUNT: u64 = 1000000;
 
+    /// The initial balance of the Subsidy fund in Mist (1 Billion * 10^9)
+    const INIT_STAKE_SUBSIDY_FUND_BALANCE: u64 = 1000000000000000000;
```
[the remark this answers is not in the document]
Comment (bmwill) [reply crates/sui-framework/sources/governance/genesis.move:31]: Wait this works?! I wanted to but i went looking for examples elsewhere in the code and couldn't find any so i assumed you couldn't, guess i should have just tried lol

File: crates/sui-framework/sources/governance/stake_subsidy.move [hunk 3]
```diff
@@ -23,44 +25,34 @@ module sui::stake_subsidy {
...
-        balance::join(
-            &mut subsidy.balance, 
-            balance::increase_supply(supply, (amount_to_mint as u64))
-        );
-    }
+    /// Advance the epoch counter and draw down the subsidy for the epoch.
+    public(friend) fun advance_epoch(subsidy: &mut StakeSubsidy): Balance<SUI> {
+        // Take the minimum of the reward amount and the remaining balance in
+        // order to ensure we don't overdraft the remaining stake subsidy
+        // balance
```
Comment (tzakian) [inline crates/sui-framework/sources/governance/stake_subsidy.move:40]: super mega nit that you should feel fine ignoring if you want: could `balance` live on the line above (we aren't really keeping to 80 char limits), and could you add an ending period?

File: crates/sui-framework/sources/sui.move [hunk 4]
```diff
@@ -5,19 +5,33 @@
...
-    use sui::tx_context::TxContext;
-    use sui::balance::Supply;
+    use sui::tx_context::{Self, TxContext};
+    use sui::balance::{Self, Balance};
     use sui::transfer;
     use sui::coin;
 
     friend sui::genesis;
 
+    const EAlreadyMinted: u64 = 0;
```
Comment (tzakian) [inline crates/sui-framework/sources/sui.move:15]: Nit: Might be worth adding a doc comment to this constant?

**Review by tzakian — APPROVED** [review]
Looks great! Had a couple super-nitty comments on comments, but you should feel free to ignore them if it makes things faster/easier :)

**Pull request merged**